lib/Serialization/Serialization.cpp

```diff
@@ -2697,22 +2697,26 @@ void Serializer::writeForeignErrorConvention(const ForeignErrorConvention &fec){
 /// - \p decl is declared in an extension of a type that depends on
 ///   \p problemContext
 static bool contextDependsOn(const NominalTypeDecl *decl,
-                             const ModuleDecl *problemModule) {
-  return decl->getParentModule() == problemModule;
+                             const DeclContext *problemContext) {
+  const DeclContext *dc = decl;
+  do {
+    if (dc == problemContext)
+      return true;
+    if (auto *extension = dyn_cast<ExtensionDecl>(dc))
+      dc = extension->getSelfNominalTypeDecl();
```

Comment:
Ah, this is a behavior change because now it *doesn't* check if the *extension* is in `problemContext`.

Reply:
I can fix it but I need to extract the test case from SwiftPM first.